src/mongo/db/pipeline/accumulator_percentile.cpp

```diff
@@ -0,0 +1,186 @@
+
+    // Determining 'digest_size'
+    if (this->digest_size == 0){
+        
+        if (input.getDocument()["digest_size"].missing()){
+            this->digest_size = 1000;
+            }
+        else
+            {
+            this->digest_size = input.getDocument()["digest_size"].getDouble();
+            }
+    }
+
+    // ToDo: error codes are not accurate. Set better numbers later
+    // ToDo: It might be better evaluations for this part.
+    uassert(6677, "The 'perc' should be present in the input document.",
+    !input.getDocument()["perc"].missing());
+
+    uassert(6678, "The 'value' should be present in the input document.",
+    !input.getDocument()["value"].missing());
+
+    this->perc_val = input.getDocument()["perc"].getDouble() / 100;  // Converting Percentile to Quantile - [0:100] to [0:1]
+
+    // ToDo: Choose a better name for perc_input and refactor later
+    Value perc_input = input.getDocument()["value"];
+
+
+    if (merging) {
+        // We expect an object that contains both a subtotal and a count. Additionally there may
+        // be an error value, that allows for additional precision.
+        // 'input' is what getValue(true) produced below.
+        verify(perc_input.getType() == Object);
+        // We're recursively adding the subtotal to get the proper type treatment, but this only
+        // increments the count by one, so adjust the count afterwards. Similarly for 'error'.
+        processInternal(perc_input[subTotalName], false);
+        _count += perc_input[countName].getLong() - 1;
+        Value error = perc_input[subTotalErrorName];
+        if (!error.missing()) {
+            processInternal(error, false);
+            _count--;  // The error correction only adjusts the total, not the number of items.
+        }
+        return;
+    }
+
+    // ToDo: Not sure 1) Is it important for TDigest to distinguish?  2) Is it important for MongoDB to distinguish?
+    // ToDo: Going to cover all Decimal, Long and Double as a temporary, need to decide on this.
+    switch (perc_input.getType()) {
+
+        case NumberDecimal:
+        case NumberLong:
+        case NumberInt:
+        case NumberDouble:
+            values.push_back(perc_input.getDouble());
+            break;
+        default:
+            dassert(!perc_input.numeric());
+            return;
+    }
+    _count++;
+
+    if (values.size() == this->chunk_size){
+        _add_to_tdigest(values);
+        }
+    else
+        return;
+}
+
+intrusive_ptr<Accumulator> AccumulatorPercentile::create(
+    const boost::intrusive_ptr<ExpressionContext>& expCtx) {
+    return new AccumulatorPercentile(expCtx);
+}
+
+Decimal128 AccumulatorPercentile::_getDecimalTotal() const {
+    return _decimalTotal.add(_nonDecimalTotal.getDecimal());
+}
+
+Value AccumulatorPercentile::getValue(bool toBeMerged) {
+
+    // To add remainders left over a chunk
+    if (values.size() > 0){
```

Comment:
What do you think to use
`if (not values.empty()) {`

Reply:
Changed